providers/implementations/signature/dsa_sig.c

```diff
@@ -454,7 +456,7 @@ static int dsa_get_ctx_params(void *vpdsactx, OSSL_PARAM *params)
 
     p = OSSL_PARAM_locate(params, OSSL_SIGNATURE_PARAM_ALGORITHM_ID);
     if (p != NULL
-        && !OSSL_PARAM_set_octet_string(p, pdsactx->aid, pdsactx->aid_len))
+        && !OSSL_PARAM_set_octet_string(p, pdsactx->aid_buf, pdsactx->aid_len))
```

Comment:
This actually results in a subtle change of behaviour in the case where the aid has not been set yet. In that case, previously, `pdsactx->aid` would be  NULL, and this function call results in an error return, and an `ERR_R_PASSED_NULL_PARAMETER` error being raised. After this change the call is successful and sets the parameter to a zero length string.

To maintain consistency with the previous behaviour I would do this:

```suggestion
        && !OSSL_PARAM_set_octet_string(p, pdsactx->aid_len == 0 ? NULL : pdsactx->aid_buf,
                                        pdsactx->aid_len))
```

Same comment for the other similar locations.